apps/server/src/modules/files/uc/sync-files-metadata.service.ts

```diff
@@ -0,0 +1,92 @@
+/* istanbul ignore file */
+
+import { ObjectId } from '@mikro-orm/mongodb';
+import { Injectable } from '@nestjs/common';
+import { FileRecord } from '@shared/domain';
+import { FileRecordRepo } from '@shared/repo';
+import { SyncFilesRepo } from '../repo/sync-files.repo';
+import { SyncFileItem, SyncTargetFile } from '../types';
+
+@Injectable()
+export class SyncFilesMetadataService {
+	constructor(private readonly fileRecordRepo: FileRecordRepo, private readonly syncFilesRepo: SyncFilesRepo) {}
+
+	public async syncMetaData(item: SyncFileItem): Promise<SyncFileItem> {
+		if (item.target) {
+			return this.updateFileRecord(item);
+		}
+
+		return this.createFileRecord(item);
+	}
+
+	public async updateFileRecord(item: SyncFileItem): Promise<SyncFileItem> {
+		const { source } = item;
+
+		if (item.target) {
```

Comment:
if (item.target) {
	return this.updateFileRecord(item);
}  is checked double.

Reply:
I see it is only to make ts lucky...but for this a deeper if else is also bad. To add target infos seperatly like 

if (item.target) {
return this.updateFileRecord(item, item.target);
}

will work for ts without additional if else. But we should figure out why ts can not resolve it correctly over multiple functions and parms definitions.
Define a new type without target? work also. 

But all solutions are really bad.